core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-store/dot-pages.store.ts

```diff
@@ -0,0 +1,128 @@
+import { Injectable } from '@angular/core';
+import { ComponentStore, tapResponse } from '@ngrx/component-store';
+import { forkJoin, Observable } from 'rxjs';
+import { switchMap, take } from 'rxjs/operators';
+import { DotCMSContentlet } from '@dotcms/dotcms-models';
+import { DotCurrentUserService } from '@dotcms/app/api/services/dot-current-user/dot-current-user.service';
+import { DotCurrentUser } from '@dotcms/app/shared/models/dot-current-user/dot-current-user';
+import {
+    DotESContentService,
+    ESOrderDirection
+} from '@dotcms/app/api/services/dot-es-content/dot-es-content.service';
+import { ESContent } from '@dotcms/app/shared/models/dot-es-content/dot-es-content.model';
+import { DotHttpErrorManagerService } from '@dotcms/app/api/services/dot-http-error-manager/dot-http-error-manager.service';
+import { HttpErrorResponse } from '@angular/common/http';
+
+export interface DotPagesState {
+    favoritePages: {
+        items: DotCMSContentlet[];
+        total: number;
+    };
+    loggedUserId: string;
+}
+
+const FAVORITE_PAGES_ES_QUERY = `+contentType:dotFavoritePage +deleted:false +working:true`;
+
+@Injectable()
+export class DotPageStore extends ComponentStore<DotPagesState> {
+    private initialFavoritePagesLimit = 5;
+    private favoritePagesTotalSize = 0;
+    private favoritePages = [];
+
+    readonly vm$ = this.state$;
+
+    /** A function that updates the Favorite Pages Items in the state of the store.
+     * @param state DotPagesState
+     * @param categories DotCMSContentlet[]
+     * @memberof DotPageStore
+     */
+    readonly setFavoritePages = this.updater<DotCMSContentlet[]>(
+        (state: DotPagesState, favoritePages: DotCMSContentlet[]) => {
+            return {
+                ...state,
+                favoritePages: {
+                    ...state.favoritePages,
+                    items: [...favoritePages]
+                }
+            };
+        }
+    );
+
+    // EFFECTS
+    readonly loadAllFavoritePages = this.effect<void>((trigger$: Observable<unknown>) =>
```

Comment:
Is the `<void>` required? first time I see it.

Reply:
yes it's needed, since an effect is being called with No parameters

Reply:
with refactor, we only have 1 function to load data, so this comment does not apply anymore